specification/graphrbac/data-plane/stable/1.6/graphrbac.json

```diff
@@ -148,6 +148,104 @@
         }
       }
     },
+    "/{tenantID}/deletedApplications/{objectId}/restore": {
+      "post": {
+        "tags": [
+          "deletedApplications"
+        ],
+        "operationId": "DeletedApplications_Restore",
+        "description": "Restores the deleted application in the directory.",
+        "parameters": [
+    	  {
+            "name": "objectId",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "Application object ID."
+          },
+          {
+            "$ref": "#/parameters/tenantIDInPath"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The application was restored successfully.",
+            "schema": {
+              "$ref": "#/definitions/Application"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/GraphError"
+            }
+          }
+        }
+      }
+    },
+    "/{tenantID}/deletedApplications": {
+      "get": {
+        "tags": [
+          "deletedApplications"
+        ],
+        "operationId": "DeletedApplications_Get",
+        "description": "Gets a list of deleted applications in the directory.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/tenantIDInPath"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The applications were retrieved successfully.",
+            "schema": {
+              "$ref": "#/definitions/ApplicationListResult"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/GraphError"
+            }
+          }
+        }
+      }
+    },
+    "/{tenantID}/deletedApplications/{applicationObjectId}": {
+      "delete": {
+        "tags": [
+          "Application"
+        ],
+        "operationId": "Applications_HardDelete",
```

Comment:
If I understand correctly the purpose of this operation, the idea is to "purge" deletedApplications and remove for real the application without waiting 30 days. This means the app has to have been deleted before.
If that so, then operationId should be `DeletedApplications_Delete`